src/main/java/com/dampcake/bencode/BencodeOutputStream.java

```diff
@@ -126,13 +126,13 @@ public void writeDictionary(final Map<?, ?> m) throws IOException {
     private static String encode(final String s) {
         if (s == null) throw new NullPointerException("s cannot be null");
 
-        return String.format("%d%s%s", s.length(), Bencode.SEPARATOR, s);
+        return String.format("%s%s%s", Integer.toString(s.length()), Bencode.SEPARATOR, s);
     }
 
     private static String encode(final Number n) {
         if (n == null) throw new NullPointerException("n cannot be null");
 
-        return String.format("%s%d%s", Bencode.NUMBER, n.longValue(), Bencode.TERMINATOR);
+        return String.format("%s%s%s", Bencode.NUMBER, Long.toString(n.longValue()), Bencode.TERMINATOR);
```

Comment:
Bencode states that numbers should be stored in base10. When would n.longValue (a long) need to be converted to a string? This seems like something that should be done on the client side.